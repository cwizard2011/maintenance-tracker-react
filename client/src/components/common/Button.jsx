import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { value, requestId } = props;
  return (
    <div className="center">
      <Link to={`/request/${requestId}/edit`}>
        <a role="presentation" className="btn center">
          <ion-icon name="create" />
          {value}
        </a>
      </Link>
    </div>
  );
};
Button.propTypes = {
  value: PropTypes.string,
  requestId: PropTypes.string
};
export default Button;
