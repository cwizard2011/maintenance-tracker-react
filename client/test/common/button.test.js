import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../src/components/common/Button';

describe('Guest Navigation', () => {
  it('render guest navigation correctly', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});
