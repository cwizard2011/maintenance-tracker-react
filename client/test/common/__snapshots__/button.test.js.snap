// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest Navigation render guest navigation correctly 1`] = `
<div
  className="center"
>
  <Link
    replace={false}
    to="/request/undefined/edit"
  >
    <a
      className="btn center"
      role="presentation"
    >
      <ion-icon
        name="create"
      />
    </a>
  </Link>
</div>
`;
